### ISSUE TITLE
allocator passed in to init()

### ISSUE THREAD

**Issue by komuw** [clownpriest/strings#1]
Hi, 

Thanks for this library, it's exactly what I have been looking for. 
Since in zig, most functions/structs etc that can allocate usually take in an allocator as the first parameter, 
wouldn't it make more sense then for the string struct to take an aloocator in it's `init` method?
https://github.com/clownpriest/strings/blob/9e55cbc8b33ddd42ebf945ae4f3f1ed097a7d758/src/strings.zig#L17

**Comment by clownpriest** [comment]
yeah I originally wrote it that way, then tried to make it more "user friendly" by just handling allocation without users having to worry about it. but it should probably be changed to require an allocator in the init(). it just feels more heavy to have to pass an allocator to  an init() for something that should be as lightweight as a string. also, strings require pretty specialized allocation patterns and maybe the library should be more opinionated about how that's done? idk....
